pkg/apis/cr/v1alpha1/types.go

```diff
@@ -257,6 +258,7 @@ type Location struct {
 	Endpoint string       `json:"endpoint"`
 	Prefix   string       `json:"prefix"`
 	Region   string       `json:"region"`
+	Hostname string       `json:"hostname"`
```

Comment:
Could this go into the credential as well? That way, the location could stay the same